pkg/k8s/client.go

```diff
@@ -121,6 +125,23 @@ func NodeNameFilter(nodeName string) internalinterfaces.TweakListOptionsFunc {
 	}
 }
 
+// GetPod returns pod that matches namespace and name
+func (c *KubeClient) GetPod(namespace, name string) (v1.Pod, error) {
+	// TODO (aramase) wrap this with retries
```

Comment:
Shouldn't we do this before merging since we see this happen in regular operation ?

Reply:
I don't think this is a blocker for merging the PR. I can add retries in follow up if required.